src/caencoder.c

```diff
@@ -278,10 +281,49 @@ static int ca_encoder_node_read_device_size(CaEncoderNode *n) {
         if (n->fd < 0)
                 return -EBADFD;
 
-        if (ioctl(n->fd, BLKGETSIZE, &u) < 0)
+        r = read(n->fd, &magic, sizeof(magic));
+        if (r != sizeof(magic))
+                return -EBADFD;
+
+        switch(le32toh(magic)) {
+                case SQUASHFS_MAGIC: {
+                        struct squashfs_super_block {
+                                /* unsigned int   s_magic; */
+                                unsigned int   inodes;
+                                int            mkfs_time;
+                                unsigned int   block_size;
+                                unsigned int   fragments;
+                                unsigned short compression;
+                                unsigned short block_log;
+                                unsigned short flags;
+                                unsigned short no_ids;
+                                unsigned short s_major;
+                                unsigned short s_minor;
+                                long long      root_inode;
+                                long long      bytes_used;
+                                /* ignore the rest */
+                        } __attribute__ ((packed)) sblk;
```

Comment:
could you add a `_packed_` macro to util.h next to the others, and use this here?